server/load_store.go

```diff
@@ -73,52 +112,8 @@ func LoadStoreRouter(ctx context.Context, cfg CLIConfig, log log.Logger) (*store
 	}
 
 	// determine read fallbacks
-	fallbacks := make([]store.PrecomputedKeyStore, len(cfg.EigenDAConfig.FallbackTargets))
-
-	for i, f := range cfg.EigenDAConfig.FallbackTargets {
-		b := store.StringToBackendType(f)
-
-		switch b {
-		case store.S3:
-			fallbacks[i] = s3
-
-		case store.EigenDA, store.Memory:
-			return nil, fmt.Errorf("EigenDA cannot be used as a fallback target")
-
-		case store.Redis:
-			return nil, fmt.Errorf("redis is not supported yet")
-
-		case store.Unknown:
-			fallthrough
-
-		default:
-			panic(fmt.Sprintf("Unknown fallback target: %s", f))
-		}
-	}
-
-	// determine caches for priority reads
-	caches := make([]store.PrecomputedKeyStore, len(cfg.EigenDAConfig.CacheTargets))
-
-	for i, f := range cfg.EigenDAConfig.CacheTargets {
-		b := store.StringToBackendType(f)
-
-		switch b {
-		case store.S3:
-			caches[i] = s3
-
-		case store.EigenDA, store.Memory:
-			return nil, fmt.Errorf("EigenDA cannot be used as a cache target")
-
-		case store.Redis:
-			return nil, fmt.Errorf("redis is not supported yet")
-
-		case store.Unknown:
-			fallthrough
-
-		default:
-			log.Warn("Unknown fallback target", "target", f)
-		}
-	}
+	fallbacks := populateTargets(cfg.EigenDAConfig.FallbackTargets, s3, redis)
+	caches := populateTargets(cfg.EigenDAConfig.CacheTargets, s3, redis)
```

Comment:
Ah that makes sense. So I think the main discrepancy is that a "normal" cache (in other circumstances) has write-through/write-back modes because we write to the cache, whereas in our case we never write to the cache (alone) right?

For us defn flow is always:
For writes:
   1. write to eigenda (or memstore)
   2. write to cache and fallback
For reads:
   1. read from cache, if not present
   2. read from eigenda/memstore, if not present
   3. read from fallback

Right?

Reply:
That's correct with the exception that step 4 will always be present. also for step 3 did you mean if present?

Reply:
weird wording, I meant it as a suffix, as in "read from cache... if not present, then go to next step"
step 4 can not be present right? isn't that the whole point of a fallback? that if eigenda is not present, then you go to the fallback?

Reply:
gonna regurgitate this using different nomenclature but it should be identical to what you're saying:

**Writing**
1. Write to EigenDA backend (or memstore)
2. Write to cache and fallback targets

**Reading**
1. Read from cache if present (i.e, len(caches) > 0)
2. Read from eigenda/memstore 
3. If step 2 failed && len(fallbacks) > 0 --> read from fallbacks

Reply:
This is awesome. Can we add this somewhere in the code as a comment?

Reply:
It already is: https://github.com/Layr-Labs/eigenda-proxy/blob/main/store/router.go#L76-L97

Reply:
ggwp